pkgs/development/tools/analysis/qcachegrind/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchurl, bash, cmake, qt57, graphviz,  qmakeHook, makeQtWrapper, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "kcachegrind-${version}";
+  version = "16.12.2";
+
+  src = fetchurl {
+    url = "http://download.kde.org/stable/applications/${version}/src/${name}.tar.xz";
+    sha256 = "0wcizav1z3n31n7qcc05mx42vl2rv71c7haaa8vn0ma39rj4iaqv";
+  };
+
+  buildInputs = [ qt57.qtbase graphviz ];
+  nativeBuildInputs = [ qmakeHook makeQtWrapper makeWrapper ];
+
+  postInstall = stdenv.lib.optionalString (stdenv.isDarwin) ''
```

Comment:
I could separate `postInstall` to build a `.desktop` for linux and an `.app` for macos. Let me known what you think.